### PR TITLE
Force pipeline barriers between unordered usages

### LINKED ISSUE #272: Pipeline barriers without memory barriers
Currently, the logic of placing barriers is simple: if a resource changes the usage, we insert a barrier that transitions this resource. Problem is - it doesn't handle a case where multiple consecutive compute commands use the same resource as `STORAGE` (or transfer commands use the same resource as destination, with ranges overlapping).

In this case, we'd need to issue a pipeline barrier that doesn't change the state of a resource.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #272

### CODE REVIEW

File: wgpu-native/src/resource.rs [hunk 1]
```diff
@@ -31,7 +31,15 @@ bitflags! {
         const STORAGE = 128;
         const INDIRECT = 256;
         const NONE = 0;
-        const WRITE_ALL = 2 + 8 + 128;
+        /// The combination of all read-only usages.
```
[the remark this answers is not in the document]
Comment (kvark) [reply wgpu-native/src/resource.rs]: yep, it's fine :)

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
Bors retry

**Comment by grovesNL** [comment]
In Travis we get

    make: *** No rule to make target 'examples-native'.  Stop.

on the Windows nightly target, which seems correct (we removed it a while ago) and we need to update the .travis.yml.

I'm not sure how the other merges were ok though, for example during the merge for #282 we get

    make: Nothing to be done for 'examples-native'.

Can make targets be cached even if the target has been removed from the Makefile?

**Comment by kvark** [comment]
bors r=grovesNL

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors r=grovesNL

**Pull request merged**